pyramid/tier_fs.go

```diff
@@ -242,35 +242,33 @@ func (tfs *TierFS) openFile(fileRef localFileRef, fh *os.File) (*ROFile, error)
 // and places it in the local FS for further reading.
 // It returns a file handle to the local file.
 func (tfs *TierFS) readFromBlockStorage(fileRef localFileRef) (*os.File, error) {
-	var e error
-	tfs.keyLock.Lock(fileRef.filename, func() {
+	_, err := tfs.keyLock.Compute(fileRef.filename, func() (interface{}, error) {
+		var err error
```

Comment:
do we need this one?

Reply:
Alas, yes.  The last error handler on l. 264 is nonfatal (not sure why; @itaiad200 can explain) and returned only after being reported to the histograms.  So we need to generate an actual `err` here.